applications/main/nfc/plugins/supported_cards/ndef.c

```diff
@@ -0,0 +1,484 @@
+        uint16_t field_len = bit_lib_bytes_to_num_be(payload + i, 2);
+        i += 2;
+
+        if(field_id == CREDENTIAL_FIELD_ID) {
+            furi_string_cat(str, "WiFi\n");
+            size_t start_position = i;
+            while(i < start_position + field_len) {
+                uint16_t cfg_id = bit_lib_bytes_to_num_be(payload + i, 2);
+                i += 2;
+                uint16_t cfg_len = bit_lib_bytes_to_num_be(payload + i, 2);
+                i += 2;
+
+                if(i + cfg_len > start_position + field_len) {
+                    return;
+                }
+
+                switch(cfg_id) {
+                case SSID_FIELD_ID:
+                    print_data(str, "SSID", payload + i, cfg_len, false);
+                    i += cfg_len;
+                    break;
+                case NETWORK_KEY_FIELD_ID:
+                    if(cfg_len > MAX_NETWORK_KEY_SIZE_BYTES) {
+                        return;
+                    }
+                    print_data(str, "PWD", payload + i, cfg_len, false);
+                    i += cfg_len;
+                    break;
+                case AUTH_TYPE_FIELD_ID:
+                    if(cfg_len != AUTH_TYPE_EXPECTED_SIZE) {
+                        return;
+                    }
+                    short auth_type = bit_lib_bytes_to_num_be(payload + i, 2);
+                    i += 2;
+                    const char* auth;
+                    switch(auth_type) {
+                    case AUTH_TYPE_OPEN:
+                        auth = "Open";
+                        break;
+                    case AUTH_TYPE_WPA_PSK:
+                        auth = "WPA Personal";
+                        break;
+                    case AUTH_TYPE_WPA_EAP:
+                        auth = "WPA Enterprise";
+                        break;
+                    case AUTH_TYPE_WPA2_EAP:
+                        auth = "WPA2 Enterprise";
+                        break;
+                    case AUTH_TYPE_WPA2_PSK:
+                        auth = "WPA2 Personal";
+                        break;
+                    case AUTH_TYPE_WPA_AND_WPA2_PSK:
+                        auth = "WPA/WPA2 Personal";
+                        break;
+                    default:
+                        auth = "Unknown";
+                        break;
+                    }
+                    print_data(str, "AUTH", (uint8_t*)auth, strlen(auth), false);
+                    break;
+                default:
+                    i += cfg_len;
+                    break;
+                }
+            }
+            return;
+        }
+        i += field_len;
+    }
+}
+
+static void parse_ndef_payload(
+    FuriString* str,
+    uint8_t tnf,
+    const char* type,
+    uint8_t type_len,
+    const uint8_t* payload,
+    uint32_t payload_len) {
+    if(!payload_len) {
+        furi_string_cat(str, "Empty\n");
+        return;
+    }
+    switch(tnf) {
+    case 0x01: // NFC Forum well-known type [NFC RTD]
```

Comment:
It would be better to replace all these numbers with appropriate defines and remove comments at all.